File: python/examples/error_handling.py
from twofinance_sdk_client import HTTPError, TwoFinanceClient, config_from_env


def main() -> None:
    client = TwoFinanceClient(config_from_env())

    try:
        client.analytics.indicators()
    except HTTPError as error:
        print(f"request failed with status {error.status_code}: {error.body}")


if __name__ == "__main__":
    main()

File: javascript/examples/error-handling.js
"use strict";

const { ServiceError, TwoFinanceClient, configFromEnv } = require("../src");

async function main() {
  const client = new TwoFinanceClient(configFromEnv(process.env));

  try {
    await client.analytics.indicators();
  } catch (error) {
    if (error instanceof ServiceError) {
      console.log(`request failed with status ${error.statusCode}: ${error.body}`);
      return;
    }
    throw error;
  }
}

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});
